common/static/common/js/karma.common.conf.js

```diff
@@ -0,0 +1,210 @@
+    return "Javascript." + browser.name.split(" ")[0];
+}
+
+
+/**
+ * Return array containing default and user supplied reporters
+ * @param {Object} config
+ * @return {Array}
+ */
+function reporters(config) {
+    var defaultReporters = ['dots', 'junit', 'kjhtml'];
+    if (config.coverage) {
+        defaultReporters.push('coverage');
+    }
+    return defaultReporters;
+}
+
+
+/**
+ * Split a filepath into basepath and filename
+ * @param {String} filepath
+ * @return {Object}
+ */
+function getBasepathAndFilename(filepath) {
+    if (!filepath) {
+        // these will configure the reporters to create report files relative to this karma config file
+        return {
+            dir: undefined,
+            file: undefined
+        };
+    }
+
+    var file = filepath.replace(/^.*[\\\/]/, ''),
+        dir = filepath.replace(file, '');
+
+    return {
+        dir: dir,
+        file: file
+    };
+}
+
+
+/**
+ * Return coverage reporter settings
+ * @param {String} config
+ * @return {Object}
+ */
+function coverageSettings(config) {
+    var path = getBasepathAndFilename(config.coveragereportpath);
+    return {
+        dir: path.dir,
+        subdir: '.',
+        reporters: [
+            {type: 'cobertura', file: path.file},
+            {type: 'text-summary'}
+        ]
+    };
+}
+
+
+/**
+ * Return junit reporter settings
+ * @param {String} config
+ * @return {Object}
+ */
+function junitSettings(config) {
+    var path = getBasepathAndFilename(config.junitreportpath);
+    return {
+        outputDir: path.dir,
+        outputFile: path.file,
+        suite: 'javascript',
+        useBrowserName: false,
+        nameFormatter: junitNameFormatter,
+        classNameFormatter: junitClassNameFormatter
+    };
+}
+
+var getConfig = function (config, useRequireJs) {
+    useRequireJs = useRequireJs === undefined ? true : useRequireJs;
+
+    var getFrameworkFiles = function () {
+        var files = [
+            'node_modules/jquery/dist/jquery.js',
+            'node_modules/jasmine-core/lib/jasmine-core/jasmine.js',
+            'node_modules/karma-jasmine/lib/boot.js',
+            'node_modules/karma-jasmine/lib/adapter.js',
+            'node_modules/jasmine-jquery/lib/jasmine-jquery.js'
+        ];
+
+        if (useRequireJs) {
+            files = files.concat([
+                'node_modules/requirejs/require.js',
+                'node_modules/karma-requirejs/lib/adapter.js'
+            ]);
+        }
+
+        return files;
+    };
+
+    var initFrameworks = function (files) {
+        getFrameworkFiles().reverse().forEach(function (f) {
+            files.unshift({
+                pattern: path.join(appRoot, f),
+                included: true,
+                served: true,
+                watch: false
+            });
+        });
+    };
+
+    initFrameworks.$inject = ['config.files'];
+
+    var customPlugin = {
+        'framework:custom': ['factory', initFrameworks]
+    };
```

Comment:
@ehteshamkafeel Can you add some documentation for `getFrameworkFiles`, `initFrameworks`, `initFrameworks.$inject` and `customPlugin`.  Like why we are doing all the above? This will help to understand this whole configuration and will also help future contributors.